hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientSpi.java

```diff
@@ -178,9 +178,9 @@ public static IOException getIOExceptionForSendCommand(
    * @return reply containing the min commit index replicated to all or majority
    *         servers in case of a failure
    */
-  public abstract XceiverClientReply watchForCommit(long index)
-      throws InterruptedException, ExecutionException, TimeoutException,
-      IOException;
+  public CompletableFuture<XceiverClientReply> watchForCommit(long index) {
```

Comment:
Shall we also change this one to `watchForCommitAsync` without breaking compatibility?